packages/flutter/lib/src/foundation/memory_allocations.dart

```diff
@@ -18,6 +18,39 @@ const bool _kMemoryAllocations = bool.fromEnvironment('flutter.memory_allocation
 /// `--dart-define=flutter.memory_allocations=true`.
 const bool kFlutterMemoryAllocationsEnabled = _kMemoryAllocations || kDebugMode;
 
+/// Create an object with exemption from release.
+///
+/// All objects created by [builder] will be exempt from release.
+///
+/// Exempt from release for an object means the object is
+/// not expected to be disposed and garbage collected
+/// during application lifecycle. However, if the object is
+/// garbage collected without being disposed, it is considered as
+/// violation of disposal contract even for exempted from release
+/// objects.
+///
+/// Noop if [kFlutterMemoryAllocationsEnabled] is false.
+///
+/// The method is useful for creating singletons:
+///
+/// ```dart
+/// class MyDisposableClass {}
```

Comment:
This should likely just be in an "Examples can assume" section to clean up this example a little bit.

Reply:
I would be happy to. I am getting doc error from bot, that is saying all examples must compile. Am I treating it wrong?

Reply:
Removed to check what bot will say.

Reply:
I don't think you can just remove it. As I said in my previous comment, it will need to go in a "Examples can assume" section. Search the repository for that string to see examples.

Reply:
Interesting! Added. Thanks.